osgp/protocol-adapter-dlms/osgp-dlms/src/main/java/org/opensmartgridplatform/dlms/services/ObjectConfigService.java

```diff
@@ -235,4 +239,54 @@ private DlmsProfile getDlmsProfile(
   public List<DlmsProfile> getConfiguredDlmsProfiles() {
     return this.dlmsProfiles;
   }
+
+  public static List<String> getCaptureObjectDefinitions(final CosemObject profile) {
+    if (profile.getClassId() != InterfaceClass.PROFILE_GENERIC.id()) {
+      throw new IllegalArgumentException(
+          "Can't get capture objects, object " + profile.getTag() + " is not a Profile Generic");
+    }
+
+    final Attribute captureObjectsAttribute =
+        profile.getAttribute(ProfileGenericAttribute.CAPTURE_OBJECTS.attributeId());
+
+    final String captureObjectsAttributeValue = captureObjectsAttribute.getValue();
+
+    if (captureObjectsAttributeValue.equals("EMPTY")) {
+      return List.of();
+    }
+
+    final String[] captureObjectDefinitions = captureObjectsAttribute.getValue().split("\\|");
```

Comment:
Waarom is de attribute value geen array?

Reply:
As discussed, the value of an attribute is a String. Using an array would be complicating things. Moved the splitting to the ProfileGeneric class. This class now returns a list of Capture Objects.